checks.c

```diff
@@ -1354,11 +1357,19 @@ static void CheckEKU(X509 *x509, CertType type)
 			{
 				SetCertInfo(CERT_INFO_OCSP_SIGN);
 			}
-			else 
+			else if (OBJ_cmp(oid, obj_IntelAMTvProEKU) == 0)
+			{
+				SetCertInfo(CERT_INFO_AMTVPRO_EKU);
+			}
+			else
 			{
 				SetWarning(WARN_UNKNOWN_EKU);
 			}
 		}
+		if (GetBit(cert_info, CERT_INFO_AMTVPRO_EKU) && !GetBit(cert_info, CERT_INFO_SERV_AUTH))
+		{
+			SetWarning(WARN_UNKNOWN_EKU);
+		}
```

Comment:
From the link, I think this should actually be an error.

Reply:
Agreed.  Addressed by https://github.com/robstradling/x509lint/commit/67bb394865f529df0d4349eed26af95c48190777